File: Practica2/QytetetJava/src/modeloqytetet/Jugador.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

import java.util.ArrayList;

/**
 *
 * @author Lorena
 */
public class Jugador {
    private boolean encarcelado;
    private String nombre;
    private int saldo;
    private Casilla casillaActual;
    private Sorpresa cartaLibertad=null;
    private ArrayList<TituloPropiedad> propiedades;

    //Constructores
    public Jugador(String nombre){
        this.nombre = nombre;
        encarcelado = false;
        saldo = 7500;
        casillaActual = null;
        propiedades = new ArrayList();
    }
    //Consultores y modificadores
    public Casilla getCasillaActual(){
        return casillaActual;
    }
    
    public boolean getEncarcelado(){
        return encarcelado;
    }
    
    void setCartaLibertad(Sorpresa carta){
        cartaLibertad = carta;
    }
    
    void setCasillaActual(Casilla casilla){
        casillaActual = casilla;
    }
    
    void setEncarcelado(boolean encarcelado){
        this.encarcelado = encarcelado;
    }
    
    //Métodos
    public boolean tengoPropiedades(){
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    boolean actualizarPosicion(Casilla casilla){
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    boolean comprarTitulo(){
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    Sorpresa devolverCartaLibertad(){
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    void irACarcel(Casilla casilla){
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    void modificarSaldo(int cantidad){
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    int obtenerCapital(){
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    ArrayList<TituloPropiedad> obtenerPropiedadesHipotecadas(boolean hipotecada){
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    void pagarCobrarPorCasaYHotel(int cantidad){
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    boolean pagarLibertad(int cantidad){
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    boolean puedoEdificarCasa(Casilla casilla){
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    boolean puedoEdificarHotel(Casilla casilla){
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    boolean puedoHipotecar(Casilla casilla){
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    boolean puedoPagarHipoteca(Casilla casilla){
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    boolean puedoVenderPropiedad(Casilla casilla){
        throw new UnsupportedOperationException("Sin implementar");
    }
    
   
    boolean tengoCartaLibertad(){
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    void venderPropiedad(Casilla casilla){
        
    }
    
    private int cuantasCasasHotelesTengo(){
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    private void eliminarDeMisPropiedades(Casilla casilla){
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    private boolean esDeMiPropiedad(Casilla casilla){
        throw new UnsupportedOperationException("Sin implementar");
    }
    
    private boolean tengoSaldo(int cantidad){
        throw new UnsupportedOperationException("Sin implementar");
    }
    

    
       //Método toString() devuelve un String con el estado del objeto correspondiente
    @Override
    public String toString(){
        return "Nombre: " + nombre + "\n"
                + "Está encarcelado? " + encarcelado + "\n"
                + "Saldo: " + saldo + "\n"
                + "Tiene carta libertad? " + cartaLibertad + "\n"
                + "Propiedades: " + propiedades.toString() + "\n";
    }
    
}

File: Practica1/Examen/QytetetRubyExamen17-18/lib/jugador.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'sorpresa'
require_relative 'titulo_propiedad'
require_relative 'casilla'

module ModeloQytetet
  class Jugador
    
    #Consultores y modificadores
    attr_accessor :encarcelado
    attr_reader :nombre, :saldo
    attr_writer :carta_libertad, :casilla_actual
    
    #Constructor
    def initialize(nom)
      @nombre = nom
      @encarcelado = false
      @saldo = 7500
      @casilla_actual = nil
      @propiedades = Array.new
      @carta_libertad = nil
    end
    
    #Métodos
    def tengo_propiedades()
      
    end
    
    def actualizar_posicion(casilla)
      
    end
    
    def comprar_titulo()
      
    end
    
    def devolver_carta_libertad()
      
    end
    
    def ir_a_carcel(casilla)
      
    end
    
    def modificar_saldo(cantidad)
      
    end
    
    def obtener_capital()
      
    end
    
    def obtener_propiedades_hipotecadas(hipotecada)
      
    end
    
    def pagar_cobrar_por_casa_y_hotel(cantidad)
      
    end
    
    def pagar_libertad(cantidad)
      
    end

    def puedo_edificar_casa(casilla)
      
    end
    
    def puedo_edificar_hotel(casilla)
      
    end
    
    def puedo_hipotecar(casilla)
      
    end
    
    def puedo_pagar_hipoteca(casilla)
      
    end
    
    def puedo_vender_propiedad(casilla)
      
    end
    
    def tengo_carta_libertad()
      
    end
    
    def vender_propiedad(casilla)
      
    end
    
    def cuantas_casas_hoteles_tengo()
      
    end
    
    def eliminar_de_mis_propiedades(casilla)
      
    end
    
    def es_de_mi_propiedad(casilla)
      
    end
    
    def tengo_saldo(cantidad)
      
    end
    
    #Método toString() devuelve un String con el estado del objeto correspondiente
    def to_s
      "************************************************\n
       Nombre:  #{@nombre} \n
       Esta encarcelado? #{@encarcelado} \n
       Saldo:  #{@saldo} \n
       Tiene carta libertad? #{@carta_libertad} \n
       Propiedades: #{@propiedades} \n"
    end
    
    private :cuantas_casas_hoteles_tengo, :eliminar_de_mis_propiedades, :es_de_mi_propiedad, :tengo_saldo
    
  end
end
